### ISSUE TITLE
Negative longitude arguments for instantiating a coordinate are mutated

### ISSUE THREAD

**Issue by 5tefan** [sunpy/sunpy#5115]
### Description
The `tx` and `ty` arguments to a Helioprojective coordinate frame can be mutated in an unexpected way.

### Expected vs Actual behavior
I would expect instantiating a Helioprojective coordinate frame to not mutate the `tx` and `ty` arguments. Currently, in some cases, the arguments are mutated.

### Steps to Reproduce

```python
def test_hpc_mutation():
    # Check that Helioprojective does not mutate arguments
    x_orig = -2.67507183
    y_orig = 2.61410036
    tx = x_orig * u.arcmin
    ty = y_orig * u.arcmin
    obstime = "2020-01-01"

    hpc = Helioprojective(tx, ty, observer="earth", obstime=obstime)

    assert tx.to_value(u.arcmin) == x_orig
    assert ty.to_value(u.arcmin) == y_orig
```

### System Details

System : Darwin
Processor : i386
Arch : 64bit
SunPy : 3.0.dev336+g164f57c41.d20210319
OS: Mac OS X 10.14.6 (i386)

Python: 3.9.2
NumPy: 1.20.1
SciPy: NOT INSTALLED
matplotlib: 3.3.4
Astropy: 4.2
Pandas: NOT INSTALLED
parfive: 1.2.0

**Comment by ayshih** [comment]
You didn't need to create a demo PR for this or go through the testing suite for this.  Straight output is sufficient:
```python
>>> x_orig = -2.67507183
>>> y_orig = 2.61410036
>>> tx = x_orig * u.arcmin
>>> ty = y_orig * u.arcmin
>>> obstime = "2020-01-01"
>>> tx
<Quantity -2.67507183 arcmin>

>>> hpc = Helioprojective(tx, ty, observer="earth", obstime=obstime)
>>> tx
<Quantity 21597.32492817 arcmin>
```

What's going on is that `tx` is being converted to a longitude, because that's what Helioprojective theta_x is, and being wrapped to be within 0 to 360 degrees.
```python
>>> tx - 360*u.deg
<Quantity -2.67507183 arcmin>
```

I presume the conversion is intentionally occurring in place rather than on a copy (i.e., for performance reasons), but it could be a bug.  I'll try to track it down, but I'll note that it'd be a bug in Astropy since you see it with plain Astropy as well:
```python
>>> import astropy.units as u
>>> from astropy.coordinates import ICRS
>>> test = -1*u.deg
>>> test
<Quantity -1. deg>

>>> ICRS(test, 0*u.deg)
>>> test
<Quantity 359. deg>
```

**Comment by 5tefan** [comment]
Ok. Thanks for providing more insight on this. I didn't recognize 21597.32492817 arcmin as the positive wrap around.

I've implemented a workaround in our processing, but this was causing headache because we're converting helioprojective cartesian to heliocentric radial (r, theta), using just `r = np.hypot(x, y)` and plugging in x=~21000 gives very different answer than using x=~2! Please let me know if something in here sounds absolutely wrong or there's a better way to do that. Not a solar person!

**Comment by ayshih** [comment]
Two options come to mind:

### Use the coordinates framework to perform the conversion
```python
>>> from astropy.coordinates import SkyCoord
>>> hpc = SkyCoord(tx, ty, frame='helioprojective', observer="earth", obstime=obstime)
>>> hpc.transform_to('heliocentric').cylindrical
<CylindricalRepresentation (rho, phi, z) in (AU, deg, AU)
    (0.00106489, 135.66044526, 0.0045269)>
```
This gets you very close to the standard definition of Heliocentric Radial, except `phi` is offset by 90 degrees.  Because Helioprojective is properly treated as a spherical coordinate system, this conversion is actually technically more accurate than your conversion.

### Use the `Longitude` class to wrap the `tx` from -180 degrees to 180 degrees
```python
>>> from astropy.coordinates import Longitude
>>> Longitude(21597.32492817*u.arcmin, wrap_angle=180*u.deg)
<Longitude -2.67507183 arcmin>
```
Then proceed with your conversion.

**Comment by 5tefan** [comment]
Thank you so much, that's incredibly helpful! I think we'll migrate towards the first suggestion using SkyCoord, that seems like the best solution. My workaround had been to just use the values before sending them to the frame. In any case, hopefully this issue can be resolved. It seems like something that could bite unsuspecting users!

**Comment by ayshih** [comment]
The Astropy fix has been merged and will be in Astropy 5.0, so I think we can close this issue.
